File: backend/src/main/java/com/twothree/backend/entity/Member.java
package com.twothree.backend.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

import com.twothree.backend.enums.Gender;
import com.twothree.backend.enums.MemberStatus;
import com.twothree.backend.enums.MemberRole;

@Entity
@Table(name = "members")
@Data
@EqualsAndHashCode(callSuper = true)
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Member extends BaseEntity {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(nullable = false)
    private String name;
    
    @Column(nullable = false, unique = true)
    private String email;
    
    @Column
    private String phone;
    
    @Column
    private String address;
    
    @Column
    private LocalDate birthDate;
    
    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private Gender gender;
    
    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private MemberStatus status = MemberStatus.ACTIVE;
    
    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private MemberRole role = MemberRole.MEMBER;
    
    @Column
    private String profileImageUrl;
    
    @Column
    private LocalDate baptismDate;
    
    @Column
    private LocalDate membershipDate;
    
    @Column(length = 1000)
    private String notes;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "church_id", nullable = false)
    private Church church;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "department_id")
    private Department department;
} 

File: app/lib/models/member.dart
import 'package:json_annotation/json_annotation.dart';
import '../enums/member_status.dart';
import '../enums/member_role.dart';

part 'member.g.dart';

@JsonSerializable()
class Member {
  final int? id;
  final String name;
  final String? email;
  final String? phone;
  final String? address;
  final DateTime? birthDate;
  final String? gender;
  final String? profileImageUrl;
  final int churchId;
  final int? departmentId;
  final MemberStatus status;
  final MemberRole role;
  final DateTime? createdAt;
  final DateTime? updatedAt;
  final String? createdBy;
  final String? updatedBy;

  Member({
    this.id,
    required this.name,
    this.email,
    this.phone,
    this.address,
    this.birthDate,
    this.gender,
    this.profileImageUrl,
    required this.churchId,
    this.departmentId,
    this.status = MemberStatus.active,
    this.role = MemberRole.member,
    this.createdAt,
    this.updatedAt,
    this.createdBy,
    this.updatedBy,
  });

  factory Member.fromJson(Map<String, dynamic> json) => _$MemberFromJson(json);
  Map<String, dynamic> toJson() => _$MemberToJson(this);
} 